### ISSUE TITLE
sites/default/config not writable

### ISSUE THREAD

**Issue by sarahg** [pantheon-systems/drops-8#82]
We're having an issue resolving a File System error we're seeing on the Drupal Status Report page: 
"The directory sites/default/config is not writable. You may need to set the correct directory at the file system settings page or change the current directory's permissions so that it is writable."

If we switch the site to SFTP mode in the dashboard as suggested in this issue – https://github.com/pantheon-systems/drops-8/issues/16, we can fix the error by creating the appropriate folders, but switching back to Git mode brings the error back.

**Comment by greg-1-anderson** [comment]
This is correct behavior; you must be in SFTP mode to export your configuration to the staging configuration store.

If you'd like to be able to export configuration while in git mode, you could define your own configuration store, and put it somewhere in sites/default/files/private.  However, it makes more sense vis-a-vis the development workflow for your configuration to be committed to your git repository along with your code.  To do this, you should be in SFTP mode, so that you can provide a commit comment from your Pantheon dashboard.

Leaving open for future discussion / documentation purposes.

**Comment by sarahg** [comment]
Got it, that makes sense. Thanks!

**Comment by baisong** [comment]
Just clarifying, does this mean that for all Drupal 8 pantheon sites in Git mode, for all environments, you will always see this error on the Status Report page, and that it's expected and not dangerous?

**Comment by greg-1-anderson** [comment]
Sorry, my comment above was unclear / incomplete, as I inadvertently only addressed the *behavior* without addressing the error message on the status report page.

The behavior, as I said above, is correct: you cannot write to the configuration directory if you are in git mode, so you must switch to sftp mode before you export configuration.  However, there is no way in Drupal core to export configuration; you can only download configuration.  Therefore, core should not be reporting it as an error on the status report page when this location is not writable.

There is already an issue to remove this error line in the Drupal core issue queue, so hopefully this message will disappear before the 8.0.0-stable release.  In the interim, though, yes, this message is expected and not dangerous.

**Comment by baisong** [comment]
Got it. Thank you Greg.

**Comment by greg-1-anderson** [comment]
The Drupal core issue is [2466197](https://www.drupal.org/node/2466197).
